client/components/tinymce/plugins/contact-form/plugin.js

```diff
@@ -3,18 +3,23 @@
  */
 import tinymce from 'tinymce/tinymce';
 import i18n from 'lib/mixins/i18n';
-import ReactDom from 'react-dom';
-import ReactDomServer from 'react-dom/server';
-import React from 'react';
-
+import React, { createElement } from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { createStore } from 'redux';
+import { Provider } from 'react-redux';
 /**
  * Internal Dependencies
  */
 import Gridicon from 'components/gridicon';
 import ContactFormDialog from './dialog';
+import reducer from './state/reducer';
+import { initContactForm, addDefaultField, removeField } from './state/actions';
+import { serialize, deserialize } from './shortcode-utils';
 
-const contactForm = editor => {
+const wpcomContactForm = editor => {
 	let node;
+	let store = createStore( reducer );
```

Comment:
This raises an interesting question on whether we should have separate stores in the application (particularly around "transient" data), though I'm inclined to think we should stick to having a single global app state. I actually had not taken a look at this pull request until after I submitted #2085, where I'm trying to accomplish something very similar. Specifically, you might want to reference how I'm setting a TinyMCE param during the initialization, pulled from the editor context.